fdbserver/storageserver.actor.cpp

```diff
@@ -1617,6 +1659,135 @@ ACTOR Future<Void> changeFeedPopQ(StorageServer* self, ChangeFeedPopRequest req)
+	wait(self->durableVersion.whenAtLeast(req.version + 1));
+
+	TraceEvent(SevDebug, "ServeGetCheckpointVersionSatisfied", self->thisServerID)
+	    .detail("Version", req.version)
+	    .detail("Range", req.range.toString())
+	    .detail("Format", static_cast<int>(req.format));
+
+	try {
+		std::unordered_map<UID, CheckpointMetaData>::iterator it = self->checkpoints.begin();
+		for (; it != self->checkpoints.end(); ++it) {
+			const CheckpointMetaData& md = it->second;
+			if (md.version == req.version && md.format == req.format && md.range.contains(req.range) &&
+			    md.getState() == CheckpointMetaData::Complete) {
+				req.reply.send(md);
+				TraceEvent(SevDebug, "ServeGetCheckpointEnd", self->thisServerID).detail("Checkpoint", md.toString());
+				break;
+			}
+		}
+
+		if (it == self->checkpoints.end()) {
+			req.reply.sendError(checkpoint_not_found());
+		}
+	} catch (Error& e) {
+		if (!canReplyWith(e)) {
+			throw;
+		}
+		req.reply.sendError(e);
+	}
+	return Void();
+}
+
+ACTOR Future<Void> deleteCheckpointQ(StorageServer* self, Version version, CheckpointMetaData checkpoint) {
+	wait(self->durableVersion.whenAtLeast(version));
+
+	TraceEvent("DeleteCheckpointBegin", self->thisServerID).detail("Checkpoint", checkpoint.toString());
+	state Key persistCheckpointKey(persistCheckpointKeys.begin.toString() + checkpoint.checkpointID.toString());
+	state Key pendingCheckpointKey(persistPendingCheckpointKeys.begin.toString() + checkpoint.checkpointID.toString());
+
+	try {
+		if (checkpoint.format == RocksDBColumnFamily) {
+			ASSERT(checkpoint.rocksCF.present());
+			self->checkpoints.erase(checkpoint.checkpointID);
+			const RocksDBColumnFamilyCheckpoint& rocksCF = checkpoint.rocksCF.get();
+			std::unordered_set<std::string> dirs;
+			for (const LiveFileMetaData& file : rocksCF.sstFiles) {
+				dirs.insert(file.db_path);
+			}
+			for (const std::string dir : dirs) {
+				platform::eraseDirectoryRecursive(dir);
+				TraceEvent("DeleteCheckpointRemovedDir", self->thisServerID)
+				    .detail("CheckpointID", checkpoint.checkpointID)
+				    .detail("Dir", dir);
+			}
+
+			// Remove the checkpoint record.
+		} else if (checkpoint.format == RocksDBSSTFile) {
+			throw not_implemented();
+		} else {
+			throw internal_error();
+		}
+		Version version = self->data().getLatestVersion();
+		auto& mLV = self->addVersionToMutationLog(version);
+		self->addMutationToMutationLog(
+		    mLV, MutationRef(MutationRef::ClearRange, pendingCheckpointKey, keyAfter(pendingCheckpointKey)));
+		self->addMutationToMutationLog(
+		    mLV, MutationRef(MutationRef::ClearRange, persistCheckpointKey, keyAfter(persistCheckpointKey)));
+	} catch (Error& e) {
+		// TODO: Handle errors more gracefully.
+		throw;
+	}
+
+	return Void();
+}
+
+// Serves GetFileRequests.
+ACTOR Future<Void> getFileQ(StorageServer* self, GetFileRequest req) {
+	TraceEvent("ServeGetFileBegin", self->thisServerID).detail("File", req.path).detail("Offset", req.offset);
```

Comment:
Since this is unlikely to be used for anything other thank checkpoints, it'd be nice if the TraceEvent had the word "Checkpoint" in it.

Reply:
Change applied.